src/todos/todos.service.ts

```diff
@@ -1,4 +1,40 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable } from "@nestjs/common";
+import { InjectModel } from "@nestjs/sequelize";
+import { CreateToDoDto } from "./dto/create-todo.dto";
+import { Todo } from "./todos.model";
 
 @Injectable()
-export class TodosService {}
+export class TodosService {
+  constructor(@InjectModel(Todo) private todoRepository: typeof Todo) {}
+
+  async createToDo(dto: CreateToDoDto) {
+    const todo = await this.todoRepository.create(dto);
+    return todo;
+  }
+
+  async getAllTodos() {
+    const todos = await this.todoRepository.findAll();
+    return todos;
+  }
+
+  async getTodo(id: number) {
+      
+    if (!id) return null;
+    const todo = await this.todoRepository.findOne({where: {id}});
+    return todo;
+  }
+
+  async updateTodo(id: number, dto: CreateToDoDto) {
+    
+    const toUpdate = await this.todoRepository.findOne({where: {id}});
```

Comment:
You need to rename the name of variable toUpdate.
It's not clear which data contains at this variable.

Reply:
Ok, ty